libsplinter/src/peer/mod.rs

```diff
@@ -524,8 +524,40 @@ fn add_peer(
 
     // if this is not a new peer, return success
     if new_ref_count > 1 {
-        let peer_ref = PeerRef::new(peer_id, peer_remover.clone());
-        return Ok(peer_ref);
+        if let Some(mut peer_metadata) = peers.get_by_peer_id(&peer_id).cloned() {
+            if peer_metadata.endpoints.len() == 1 && endpoints.len() > 1 {
+                // this will always be true
+                if let Some(endpoint) = peer_metadata.endpoints.get(0) {
```

Comment:
Should add in an error clause for the else then, so if code changes make it not true, its obvious.

Reply:
added